src/commands/debug_files/create_jvm_based_bundle.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs;
+use std::path::{Path};
+use anyhow::{bail, Result};
+use clap::{Arg, ArgMatches, Command};
+use sentry::types::DebugId;
+use symbolic::debuginfo::sourcebundle::{SourceFileType};
+use uuid::{Uuid};
+use crate::api::Api;
+use crate::config::Config;
+use crate::utils::args::ArgExt;
+use crate::utils::file_search::ReleaseFileSearch;
+use crate::utils::file_upload::{FileUpload, SourceFile, UploadContext};
+use crate::utils::fs::{path_as_url};
+
+pub fn make_command(command: Command) -> Command {
+    command
+        .hide(true) // experimental for now
+        .about("Create a source bundle for the given JVM based source files (e.g. Java, Kotlin, ...)")
+        .org_arg()
+        .project_arg(false)
+        .arg(
+            Arg::new("path")
+                .value_name("PATH")
+                .required(true)
+                .help("The directory containing source files to bundle."),
+        )
+        .arg(
+            Arg::new("output")
+                // .short('o')
+                .long("output")
+                .value_name("PATH")
+                .required(true)
+                .help("The path to the output folder."),
+        )
+        .arg(
+            Arg::new("debug_id")
+                // .short('d')
+                .long("debug-id")
+                .value_name("UUID")
+                .required(true)
+                .help("Debug ID to use for the source bundle."),
+        )
+}
+
+pub fn execute(matches: &ArgMatches) -> Result<()> {
+    let config = Config::current();
+    let org = config.get_org(matches)?;
+    let project = config.get_project(matches).ok();
+    let api = Api::current();
+    let chunk_upload_options = api.get_chunk_upload_options(&org)?;
+    let context = &UploadContext {
+        org: &org,
+        project: project.as_deref(),
+        release: None,
+        dist: None,
+        note: None,
+        wait: true,
+        dedupe: false,
+        chunk_upload_options: chunk_upload_options.as_ref(),
+    };
+    let path = Path::new(matches.get_one::<String>("path").unwrap());
+    let output_path = Path::new(matches.get_one::<String>("output").unwrap());
+    let debug_id_arg = matches.get_one::<String>("debug_id").unwrap();
+    let debug_id_uuid = Uuid::parse_str(debug_id_arg);
+    if debug_id_uuid.is_err() {
+        bail!("Given debug_id is invalid: {}", debug_id_arg)
+    }
+    let debug_id = DebugId::from_uuid(debug_id_uuid.unwrap());
+    let mut debug_id_string = debug_id.to_string();
+    debug_id_string.push_str(".zip");
+    let out = output_path.join(Path::new(&debug_id_string));
+
+    if !path.exists() {
+        bail!("Given path does not exist: {}", path.to_string_lossy())
+    }
```

Comment:
We could also just create the directory if it doesn't exist.

Reply:
We could for `output` but for `path` it doesn't make sense because that's the input (tree of source files) we want to bundle. Modified the code to create `output` if it doesn't exist.

Reply:
Derp, this was a misunderstanding on my part.